File: DisplayMonitorAPI/src/main/java/tororo1066/displaymonitorapi/configuration/Execute.java
package tororo1066.displaymonitorapi.configuration;

import tororo1066.displaymonitorapi.actions.IActionContext;

/**
 * {@link IActionContext}を元に処理を行う
 */
@FunctionalInterface
public interface Execute {
    /**
     * 処理を実行する
     * @param context {@link IActionContext}
     */
    void invoke(IActionContext context);

    /**
     * 空のExecuteを取得する
     * @return 空のExecute
     */
    static Execute empty() {
        return context -> {};
    }
}

File: src/main/kotlin/tororo1066/displaymonitor/elements/Execute.kt
package tororo1066.displaymonitor.elements

import tororo1066.displaymonitor.actions.ActionContext

open class Execute(val execute: (context: ActionContext) -> Unit) {
    operator fun invoke(context: ActionContext) {
        execute(context)
    }

    companion object {
        fun empty(): Execute {
            return Execute {}
        }
    }
}